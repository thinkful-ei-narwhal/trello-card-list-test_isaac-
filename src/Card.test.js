import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Card from './Card';

describe('<Card />', () => {
	//Smoke Testing
	it('Smoke Testing Card Component: render without crashing', () => {
		//DOM element to render into
		const div = document.createElement('div');

		//Render the component
		ReactDOM.render(<Card titile='title' content='content' />, div);

		//Clean up
		ReactDOM.unmountComponentAtNode(div);
	});

	//Snapshot Testing
	it('Snapshot Testing Card Component: render this UI as expected', () => {
		//Render the component as JSON
		const tree = renderer
			.create(<Card titile='title' content='content' />)
			.toJSON();

		//Check if it matches the previous snapshot
		//stored in __snapshots__/Card.test.js.snap
		expect(tree).toMatchSnapshot();
	});
});
