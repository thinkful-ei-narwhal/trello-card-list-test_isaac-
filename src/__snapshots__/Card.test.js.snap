// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Snapshot Testing Card Component: render this UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3 />
  <p>
    content
  </p>
</div>
`;
